go-controller/pkg/ovn/pods.go

```diff
@@ -374,29 +376,40 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 			return fmt.Errorf("unable to create LSPSetDynamicAddresses command for port: %s", portName)
 		}
 		cmds = append(cmds, cmd)
-	} else {
+
+		// ensure we have/can allocate the IP in the annotation
+		if err = oc.lsManager.AllocateIPs(logicalSwitch, podIfAddrs); err != nil && err != ipallocator.ErrAllocated {
+			klog.Warningf("Unable to allocate IPs for already annotated pod: %s, IPs: %s, error: %v", pod.Name,
```

Comment:
is a warning here sufficient? or do we roll-back annotations at this point?

Reply:
We cant go down the path of rolling back annotations. The reason is an annotation rollback is essentially another pod update event, which means we get an additional event. I first wrote the code this way yesterday and realized there are a lot of nuanced problems with this. @dcbw commented further down but he meant to comment here. This should be an error and return. Going to fix that.

Reply:
if you just error out here, the event will get requeued any way right? pod will error out, crashloop, send a new event yada yada yada.

Reply:
I don't think the event will be requeued. We aren't using workqueues. If someone updates the pod it will retry. Either way if we get to this point either something is seriously out of sync between KAPI and our allocator and we should just error.